### PR TITLE
Ecpw postclustering

### PR DESCRIPTION
**Pull request by ecpierce**
Main changes-
added new env variables for downstream steps
update bsoid_app.py to loop through directories for video creation
remove some unnecessary browser style setup in bsoid_app.py

### CODE REVIEW

**Comment by ecpierce** [comment]
should also note that this functions for csv/mp4 files. has not been tested on any other file types

File: bsoid_app.py
Comment (taylorreiter) [inline bsoid_app.py:60]: where does this capitalization come from?

File: bsoid_app/clustering.py
Comment (taylorreiter) [inline bsoid_app/clustering.py:61]: does it work for you to have this hardcoded or will it be annoying/have files overwritten if you run this multiple times?

File: bsoid_app/machine_learner.py
Comment (taylorreiter) [inline bsoid_app/machine_learner.py]: can you remove these extra spaces?

File: bsoid_app/machine_learner.py
Comment (taylorreiter) [inline bsoid_app/machine_learner.py]: extra spaces

File: bsoid_app/machine_learner.py
Comment (taylorreiter) [inline bsoid_app/machine_learner.py]: extra spaces

File: bsoid_app/machine_learner.py
Comment (taylorreiter) [inline bsoid_app/machine_learner.py]: extra spaces

File: bsoid_app/video_creator.py
Comment (taylorreiter) [inline bsoid_app/video_creator.py]: extra spaces

**Review by taylorreiter — COMMENTED** [review]
this lgtm but let's wait for Mert. I added a couple formatting comments and some questions I had. It might be worth running a linter on your code, although that might change A LOT of things and make the diff hard, so maybe best to wait until after this is merged in. (in the future, that could be your first PR after forking, but only if you don't plan to contribute your code back, as a PR that touches every line of code for formatting reasons would probably be annoying for someone to receive.)

File: bsoid_app/machine_learner.py
Comment (ecpierce) [reply bsoid_app/machine_learner.py]: removed

File: bsoid_app/video_creator.py
Comment (ecpierce) [reply bsoid_app/video_creator.py]: removed

File: bsoid_app.py
Comment (ecpierce) [reply bsoid_app.py:60]: this is something Mert had suggested, Protocol is a class, and I guess classes are usually capitalized in python

File: bsoid_app/machine_learner.py
Comment (ecpierce) [reply bsoid_app/machine_learner.py]: done

File: bsoid_app/machine_learner.py
Comment (ecpierce) [reply bsoid_app/machine_learner.py]: done

File: bsoid_app/machine_learner.py
Comment (ecpierce) [reply bsoid_app/machine_learner.py]: done

File: bsoid_app/machine_learner.py
Comment (mertcelebi) [inline bsoid_app/machine_learner.py]: On VSCode, hit `Command + ,` to open settings. Search for `Files: Trim Trailing Whitespace` and enable it

File: bsoid_app/export_training.py
Comment (mertcelebi) [inline bsoid_app/export_training.py:31]: You don't need this check?

File: bsoid_app.py
Comment (mertcelebi) [inline bsoid_app.py:68]: Should these be attached to the `ROOT_PATH` or `WORKING_DIR`?

Comment (mertcelebi) [reply bsoid_app.py:68]: What prompted you to do this in a loop since the original was just a single call?

File: bsoid_app/video_creator.py
Comment (mertcelebi) [inline bsoid_app/video_creator.py:24]: This seems to be missing `WORKING_DIR`?

File: bsoid_app.py
Comment (mertcelebi) [inline bsoid_app.py:70]: Why do we have to pass the `full_directory_path` twice?

File: bsoid_app.py
Comment (mertcelebi) [inline bsoid_app.py]: Do these need to be cast to `int`?

Comment (mertcelebi) [reply bsoid_app.py]: `PLAYBACK_SPEED` needs `float`

File: bsoid_app/video_creator.py
Comment (mertcelebi) [inline bsoid_app/video_creator.py:49]: So `number_example` is supposed to be int according to your documentation. But here they were using an array. Does this actually work?

File: bsoid_app/video_creator.py
Comment (mertcelebi) [inline bsoid_app/video_creator.py:174]: Lol this code is so messy (not your fault). Did you do any meaningful changes besides removing `st` stuff?

**Review by mertcelebi — CHANGES_REQUESTED** [review]
A couple questions!

File: bsoid_app.py
Comment (ecpierce) [reply bsoid_app.py:68]: ROOT_PATH is correct, this will point to the directory the video is in rather than WORKING_DIR which is the output folder. 

In the original app, the user selects an individual data file and individual video file to run video_creator. They would have to repeatedly go through this part of the app one file at a time. Here, I wanted it to just create videos for each data file

File: bsoid_app/video_creator.py
Comment (ecpierce) [reply bsoid_app/video_creator.py:49]: yes as far as I can tell this is not creating any issues 🤷‍♀️

File: bsoid_app/video_creator.py
Comment (ecpierce) [reply bsoid_app/video_creator.py:174]: main changes were getting it to create videos for all the input folders vs. the user repeatedly going through this part of the app one file at a time if they wanted it done for multiple files/videos

File: bsoid_app/export_training.py
Comment (ecpierce) [reply bsoid_app/export_training.py:31]: don't think I need this anymore because 'List of pose-relationship features' will definitely be included.    In the app, a user can select options of 'List of pose-relationship features', 'Mapping between features and assignment', 'Assignment probabilities'. Here I am just going to automatically generate all three. We could potentially make this also an option for users to provide a list of what they want, but there are already so many options it seemed unnecessary.

File: bsoid_app/video_creator.py
Comment (ecpierce) [reply bsoid_app/video_creator.py:24]: acutally don't think I need WORKING_DIR here, removed from bsoid_app.py section that passes to this

File: bsoid_app/video_creator.py
Comment (ecpierce) [reply bsoid_app/video_creator.py:49]: self.number_examples = st.slider('Select number of non-repeated examples', 1, 20, 5)

this is from the original app. You select a number from a slider, but in the end it is just an integer number of examples?

File: bsoid_app.py
Comment (ecpierce) [reply bsoid_app.py:70]: the first instance should actually pass DATA_DIRECTORIES. edited.

**Comment by ecpierce** [comment]
also added change to run_streamlit_cli.py and bsoid_app.py so that it will exit streamlit/end running after video creation

**Review by mertcelebi — APPROVED** [review]

**Pull request merged**
